src/Npgsql/TypeHandlers/NumericHandlers/NumericHandler.cs

```diff
@@ -164,13 +164,13 @@ public int ValidateAndGetLength(double value, NpgsqlParameter parameter)
         public int ValidateAndGetLength(byte value, NpgsqlParameter parameter)
             => ValidateAndGetLength((decimal)value, parameter);
 
-        public override unsafe void Write(decimal value, NpgsqlWriteBuffer buf, NpgsqlParameter parameter)
+        public override void Write(decimal value, NpgsqlWriteBuffer buf, NpgsqlParameter parameter)
         {
             var groupCount = 0;
-            var groups = stackalloc short[MaxGroupCount];
+            Span<short> groups = stackalloc short[MaxGroupCount];
             var weight = 0;
```

Comment:
I'm thinking about reordering of these fields to have better readability? Maybe make `weight` first?